scripts/main.js

```diff
@@ -0,0 +1,213 @@
+  }
+
+  const imagePreview = document.querySelector("#image-preview");
+  const reader = new FileReader();
+
+  reader.addEventListener(
+    "load",
+    () => {
+      imagePreview.src = reader.result;
+    },
+    false
+  );
+
+  reader.readAsDataURL(selectedImageFile);
+}
+
+// ---
+
+function getCaptionElement() {
+  if (state.caption) {
+    const captionElement = document.createElement("p");
+    captionElement.innerText = `"${state.caption}"`;
+    return captionElement;
+  }
+}
+
+// ---
+
+function getCopyButton() {
+  const button = document.createElement("button");
+  button.setAttribute("type", "button");
+  button.innerText = "Copy caption text";
+
+  button.addEventListener("click", onCopyButtonClicked);
+
+  return button;
+}
+
+// ---
+
+function getImgCodeSnippet() {
+  const imagePreviewElement = document.querySelector("#image-preview");
+  const imageSource = imagePreviewElement.getAttribute("src");
+  const codeSnippet = `<img alt="${state.caption}" src="${imageSource}">`;
+  return codeSnippet;
+}
+
+// ---
+
+function getCopyImgCodeSnippetButton() {
+  const button = document.createElement("button");
+  button.setAttribute("type", "button");
+  button.innerText = "Copy code snippet";
+
+  button.addEventListener("click", onCopyImgCodeSnippetButtonClicked);
+
+  return button;
+}
+
+function getImageSizeErrorModal() {
+  const modal = document.createElement('p');
+  modal.innerText = 'Image must be less than 3 MB';
+  modal.id = 'image-size-modal';
+
+  document.documentElement.addEventListener('click', (event) => {
+    onClickAfterModalAppears(event);
+  });
+
+  return modal;
+}
+
+// ---* for event listeners *---
+function onCopyImgCodeSnippetButtonClicked() {
+  navigator.clipboard.writeText(state.codeSnippet);
+}
+
+function onCopyButtonClicked() {
+  navigator.clipboard.writeText(state.caption);
+}
+
+function onImageInputChanged() {
+  if (imageInputElement.files[0].size < 3000000){
+  
+    const selectedImageFile = imageInputElement.files[0];
+    state.image = selectedImageFile.name;
+
+    renderImagePreview(selectedImageFile);
+    renderGetCaptionButton();
+    const getCaptionButton = document.querySelector("#get-caption");
+    getCaptionButton.addEventListener("click", onGetCaptionButtonClicked);
+  } else {
+    imageInputElement.value = '';
+    renderImageSizeErrorModal();
+  }
+}
+
+function onGetCaptionButtonClicked() {
+  requestForAPI();
+}
+
+function onClickAfterModalAppears(event, controller) {
+  const modal = document.getElementById('image-size-modal');
+    if (event.target !== modal) {
+      modal.remove();
+    } 
+}
+
+//---
+
+// fetch request
+async function requestForAPI() {
+  const selectedImageFile = imageInputElement.files[0];
+  const data = new FormData();
+  data.append("imageFile", selectedImageFile, "file");
+
+  const options = {
+    method: "POST",
+    body: data,
+    credentials: "omit",
+    headers: {
+      "Apikey": "eea0fd3c-6342-4543-a92c-9fc111175b97"
```

Comment:
It is best practice to never include API keys in github repos. How to do this properly is something we'll talk about more when we get into backend development & environment variables, etc. But heads up that you may want to make this repo private because of this for now (otherwise you will need to delete this commit to ensure the key is not included in the repo history, or possibly request a new key and add a "ADD_YOUR_KEY" filler to this section and instructions in the README to replace ADD_YOUR_KEY with your own private API key)